src/Native/Unix/System.Native/pal_networking.c

```diff
@@ -49,6 +52,10 @@
 #if !HAVE_IN_PKTINFO
 #include <net/if.h>
 #if HAVE_GETIFADDRS
+#if defined (__HAIKU__)
+// Haiku needs this define for getifaddrs
+#define _BSD_SOURCE
```

Comment:
Is there any way we can do this in a more generalized manner, such that any platform that needs this defined gets it, without having to name each?

Reply:
Every platform's is different.

I suppose this could be added further up to Mono's autoconf script....

Reply:
I think it'd be nice if this could be centralized to where configuration is done, such that if we have to utter "__HAIKU__", that can happen outside of the core source code.  Then when platforms need that defined, it can be done in that centralized location that's part of configuring the build, rather than having to through all of the various places that need this set throughout the source files.